src/mongocxx/test/spec/initial_dns_seedlist_discovery.cpp

```diff
@@ -0,0 +1,310 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cctype>
+#include <fstream>
+#include <mutex>
+#include <thread>
+
+#include <bsoncxx/test_util/catch.hh>
+#include <mongocxx/exception/operation_exception.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/options/pool.hpp>
+#include <mongocxx/pool.hpp>
+#include <mongocxx/test/spec/monitoring.hh>
+
+namespace {
+
+struct initial_dns_seedlist_test {
+    bsoncxx::stdx::string_view uri;
+    bsoncxx::array::view hosts;
+    bsoncxx::document::view options;
+    bool error = false;
+    bool ping = true;
+
+    static initial_dns_seedlist_test parse(bsoncxx::document::view test_doc) {
+        initial_dns_seedlist_test test;
+
+        for (auto el : test_doc) {
```

Comment:
```suggestion
        for (auto el : test_doc) {
            const auto key = el.key();
```
Avoid repeated `.key()` lookup.